OTEP.md

```diff
@@ -32,7 +32,479 @@ Using a configuration model or configuration file, users could configure all opt
 * Versioning: needs to be able to version stability while evolving
 * (stretch) The file format can be validated client side.
 
-## Internal details (TBD)
+## Internal details
+
+In order to provide a minimal API surface area, implementations *MUST* support the following methods.
+
+### LoadAndValidateConfigurationFromFile(filepath, format)
+
+An API called `LoadAndValidateConfigurationFromFile` receives a string parameter indicating the file path containing the configuration to be loaded. An optional format parameter may be provided to indicate the format that this configuration uses. The default value for this parameter is `yaml`. The method returns a `Configuration` model that has been validated. This API *MAY* return an error or raise an exception, whichever is idiomatic to the implementation for the following reasons:
+
+* file doesn't exist or is invalid
+* configuration loaded is invalid
+
+#### Python LoadAndValidateConfigurationFromFile example
+
+```python
+
+filename = "./config.yaml"
+
+try:
+  cfg = opentelemetry.LoadAndValidateConfigurationFromFile(filename)
+except Exception as e:
+  print(e)
+
+filename = "./config.ini"
+
+try:
+  cfg = opentelemetry.LoadAndValidateConfigurationFromFile(filename, format="ini")
+except Exception as e:
+  raise e
+
+```
+
+#### Go LoadAndValidateConfigurationFromFile example
+
+```go
+
+filename := "./config.yaml"
+cfg, err := otel.LoadAndValidateConfigurationFromFile(filename)
+if err != nil {
+  return err
+}
+
+filename := "./config.json"
+cfg, err := otel.LoadAndValidateConfigurationFromFile(filename, otelconfig.WithFormat("json"))
+if err != nil {
+  return err
+}
+
+```
+
+Implementations *MUST* allow users to specify an environment variable to set the configuration file. This gives flexibility to end users of implementations that do not support command line arguments. Some possible names for this variable:
+
+* `OTEL_CONFIGURATION_FILE`
+* `OTEL_CONFIG_FILE`
+* `OTEL_CONF`
```

Comment:
Probably add `_SDK_` to not to confuse? I have seen people use `OTEL_CONFIG_FILE` for collector config in their internal systems.